iothub/device/src/Transport/HttpTransportHandler.cs

```diff
@@ -252,52 +265,33 @@ public override async Task<Message> ReceiveAsync(CancellationToken cancellationT
                 return null;
             }
 
-            responseMessage.Headers.TryGetValues(CustomHeaderConstants.MessageId, out IEnumerable<string> messageId);
-            responseMessage.Headers.TryGetValues(HttpResponseHeader.ETag.ToString(), out IEnumerable<string> lockToken);
-            responseMessage.Headers.TryGetValues(CustomHeaderConstants.EnqueuedTime, out IEnumerable<string> enqueuedTime);
-            responseMessage.Headers.TryGetValues(CustomHeaderConstants.DeliveryCount, out IEnumerable<string> deliveryCountAsStr);
-            responseMessage.Headers.TryGetValues(CustomHeaderConstants.ExpiryTimeUtc, out IEnumerable<string> expiryTime);
-            responseMessage.Headers.TryGetValues(CustomHeaderConstants.CorrelationId, out IEnumerable<string> correlationId);
-            responseMessage.Headers.TryGetValues(CustomHeaderConstants.SequenceNumber, out IEnumerable<string> sequenceNumber);
-
             byte[] byteContent = await responseMessage.Content.ReadAsByteArrayAsync().ConfigureAwait(false);
 
             Message message = byteContent != null
                 ? new Message(byteContent)
                 : new Message();
 
-            message.MessageId = messageId?.First();
-            message.LockToken = lockToken?.First().Trim('\"');
-
-            if (enqueuedTime != null)
+            // Read Http headers and map them to message system properties.
+            foreach (KeyValuePair<string, IEnumerable<string>> header in responseMessage.Headers)
             {
-                if (DateTime.TryParse(enqueuedTime.First(), out DateTime enqueuedTimeUtc))
-                {
-                    message.EnqueuedTimeUtc = enqueuedTimeUtc;
-                }
-            }
+                string headerKey = header.Key;
 
-            if (deliveryCountAsStr != null)
-            {
-                if (byte.TryParse(deliveryCountAsStr.First(), out byte deliveryCount))
+                if (header.Key.Equals(HttpResponseHeader.ETag.ToString(), StringComparison.OrdinalIgnoreCase))
```

Comment:
Can we move this logic inside ConvertToMessageSystemProperty? For all other cases, it is in that method. The logic for this key would be to trim instead of parse.

Reply:
I have it outside since the Etag mapping isn't available in `CustomHeaderConstants`, but I can add this as an exception.